x-pack/plugins/ml/public/embeddables/anomaly_explorer/explorer_input_resolver.ts

```diff
@@ -0,0 +1,259 @@
+import { AnomalyDetectorService } from '../../application/services/anomaly_detector_service';
+import {
+  AnomalyExplorerEmbeddableInput,
+  AnomalyExplorerEmbeddableOutput,
+  AnomalyExplorerServices,
+} from '..';
+import type { CombinedJob } from '../../../common/types/anomaly_detection_jobs';
+import type { ExplorerChartsData } from '../../application/explorer/explorer_charts/explorer_charts_container_service';
+
+const FETCH_RESULTS_DEBOUNCE_MS = 500;
+
+function getJobsObservable(
+  embeddableInput: Observable<AnomalyExplorerEmbeddableInput>,
+  anomalyDetectorService: AnomalyDetectorService,
+  setErrorHandler: (e: Error) => void
+) {
+  return embeddableInput.pipe(
+    pluck('jobIds'),
+    distinctUntilChanged(isEqual),
+    switchMap((jobsIds) => anomalyDetectorService.getJobs$(jobsIds)),
+    catchError((e) => {
+      setErrorHandler(e.body ?? e);
+      return of(undefined);
+    })
+  );
+}
+
+export function useExplorerInputResolver(
+  embeddableInput: Observable<AnomalyExplorerEmbeddableInput>,
+  onInputChange: (output: Partial<AnomalyExplorerEmbeddableOutput>) => void,
+  refresh: Observable<any>,
+  services: [CoreStart, MlStartDependencies, AnomalyExplorerServices],
+  chartWidth: number,
+  severity: number
+): { chartsData: ExplorerChartsData; isLoading: boolean; error: Error | null | undefined } {
+  const [
+    { uiSettings },
+    { data: dataServices },
+    { anomalyTimelineService, anomalyDetectorService, anomalyExplorerService },
+  ] = services;
+  const { timefilter } = dataServices.query.timefilter;
+
+  const [chartsData, setChartsData] = useState<any>();
+  const [error, setError] = useState<Error | null>();
+  const [isLoading, setIsLoading] = useState(false);
+
+  const chartWidth$ = useMemo(() => new Subject<number>(), []);
+  const severity$ = useMemo(() => new Subject<number>(), []);
+
+  const timeBuckets = useMemo(() => {
+    return new TimeBuckets({
+      'histogram:maxBars': uiSettings.get(UI_SETTINGS.HISTOGRAM_MAX_BARS),
+      'histogram:barTarget': uiSettings.get(UI_SETTINGS.HISTOGRAM_BAR_TARGET),
+      dateFormat: uiSettings.get('dateFormat'),
+      'dateFormat:scaled': uiSettings.get('dateFormat:scaled'),
+    });
+  }, []);
+
+  useEffect(() => {
+    const subscription = combineLatest([
+      getJobsObservable(embeddableInput, anomalyDetectorService, setError),
+      embeddableInput,
+      chartWidth$.pipe(skipWhile((v) => !v)),
+      severity$,
+      refresh.pipe(startWith(null)),
+    ])
+      .pipe(
+        tap(setIsLoading.bind(null, true)),
+        debounceTime(FETCH_RESULTS_DEBOUNCE_MS),
+        switchMap(([jobs, input, embeddableContainerWidth, severityValue]) => {
+          if (!jobs) {
+            // couldn't load the list of jobs
+            return of(undefined);
+          }
+
+          const { maxSeriesToPlot, timeRange: timeRangeInput, filters, query } = input;
+
+          const viewBySwimlaneFieldName = OVERALL_LABEL;
```

Comment:
I expected it to be configurable 🤔

Reply:
For the first phase I think we are planning to keep it as `Overall` only, but this can be an input in the future.